Note:  this bug was taken from Commit-ID: 26c2d2b39128adba276d140eefa2745591b88536
and corrected for white space differences between the RHEL-7.1 3.10.0-229 upstream 
kernel.
---
diff -uNrp a/arch/x86/kernel/entry_32.S b/arch/x86/kernel/entry_32.S
--- a/arch/x86/kernel/entry_32.S	2015-05-17 07:39:14.371778177 -0500
+++ b/arch/x86/kernel/entry_32.S	2015-05-17 07:36:13.654610869 -0500
@@ -457,15 +457,14 @@ sysenter_exit:
 sysenter_audit:
 	testl $(_TIF_WORK_SYSCALL_ENTRY & ~_TIF_SYSCALL_AUDIT),TI_flags(%ebp)
 	jnz syscall_trace_entry
-	addl $4,%esp
-	CFI_ADJUST_CFA_OFFSET -4
-	movl %esi,4(%esp)		/* 5th arg: 4th syscall arg */
-	movl %edx,(%esp)		/* 4th arg: 3rd syscall arg */
-	/* %ecx already in %ecx		   3rd arg: 2nd syscall arg */
-	movl %ebx,%edx			/* 2nd arg: 1st syscall arg */
-	/* %eax already in %eax		   1st arg: syscall number */
+	/* movl PT_EAX(%esp), %eax        already set, syscall number: 1st arg to audit */
+	movl PT_EBX(%esp), %edx                /* ebx/a0: 2nd arg to audit */
+ 	/* movl PT_ECX(%esp), %ecx        already set, a1: 3nd arg to audit */
+	pushl_cfi PT_ESI(%esp)                /* a3: 5th arg */
+	pushl_cfi PT_EDX+4(%esp)        /* a2: 4th arg */
 	call __audit_syscall_entry
-	pushl_cfi %ebx
+	popl_cfi %ecx /* get that remapped edx off the stack */
+	popl_cfi %ecx /* get that remapped esi off the stack */
 	movl PT_EAX(%esp),%eax		/* reload syscall number */
 	jmp sysenter_do_call
 
---
